crypto/abi-types.go

```diff
@@ -1,4 +1,4 @@
-package state
+package crypto
```

Comment:
side note: if these were in package `abi`, the type names could be freed up from any collisions which currently require workarounds like `BytesTy`. The purpose of `abi.Bytes` is pretty clear to readers.

Reply:
I agree strongly with this.

Reply:
It wasn't clear to me whether abi types were `crypto`. They arguably are, because we need to encode and decode data as a part of our cryptographic protocol, but this feels like a weak argument.

I like the clarity that an `abi` package would bring to naming.

Reply:
0c6318e + b70f166